job-server/src/spark.jobserver/context/StreamingContextFactory.scala

```diff
@@ -0,0 +1,39 @@
+package spark.jobserver.context
+
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.Config
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+import spark.jobserver.{SparkStramingJob, SparkSqlJob, SparkJobBase, ContextLike}
+import spark.jobserver.util.SparkJobUtils
+import spark.jobserver.util.SparkJobUtils._
+
+/**
+ * Created by ZeoS on 3/18/15.
+ */
+class StreamingContextFactory extends SparkContextFactory{
+  import SparkJobUtils._
+
+  type C = StreamingContext with ContextLike
+
+  def makeContext(config: Config, contextConfig: Config, contextName: String): C = {
+    val conf = configToSparkConf(config, contextConfig, contextName)
+    val interval :Int = {
+      if (contextConfig.hasPath("streaming.batch_interval")) {
+        contextConfig.getInt("streaming.batch_interval")
+      } else {
+        if (config.hasPath("streaming.batch_interval")) {
```

Comment:
Also, you can programmatically try to get the config from "spark.context-settings" in the job config first, then fall back to the contextConfig, like this:

    val contextCfg = config.getConfig("spark.context-settings").withFallback(contextConfig)
    contextCfg.getInt("streaming.batch_interval")

Maybe the first line should be a method in SparkContextFactory, since everyone will need logic like this.